src/codegen/array-literal-expression.ts

```diff
@@ -0,0 +1,64 @@
+import llvm from 'llvm-node';
+import ts from 'typescript';
+
+import LLVMCodeGen from '.';
+
+export default class CodeGenArray {
+  private cgen: LLVMCodeGen;
+
+  constructor(cgen: LLVMCodeGen) {
+    this.cgen = cgen;
+  }
+
+  public genArrayType(node: ts.ArrayLiteralExpression): llvm.ArrayType {
+    const length = node.elements.length;
+    const elementType = (() => {
+      if (length === 0) {
+        return this.cgen.genType(
+          (this.cgen.currentType! as ts.ArrayTypeNode).elementType
+        );
+      } else {
+        return this.cgen.genExpression(node.elements[0]).type;
+      }
+    })();
+    return llvm.ArrayType.get(elementType, length);
+  }
+
+  public genArrayInitializer(node: ts.ArrayLiteralExpression): llvm.Value[] {
+    return node.elements.map(item => {
+      return this.cgen.genExpression(item);
+    });
+  }
+
+  // [0] https://stackoverflow.com/questions/38548680/confused-about-llvm-arrays
+  // [1] https://stackoverflow.com/questions/33003928/allow-llvm-generate-code-to-access-a-global-array
+  public genArrayLiteral(node: ts.ArrayLiteralExpression): llvm.AllocaInst {
+    const arrayType = this.genArrayType(node);
+    const arraySize = llvm.ConstantInt.get(
+      this.cgen.context,
+      arrayType.numElements,
+      64
+    );
+    const arrayPtr = this.cgen.builder.createAlloca(arrayType, arraySize);
+    this.genArrayInitializer(node).forEach((item, i) => {
+      const ptr = this.cgen.builder.createInBoundsGEP(arrayPtr, [
```

Comment:
Does that mean to assign `0` to every element with index `i`?

Reply:
It's a bug here, it will paniced at empty array, fixed.

Reply:
Yes. An empty sized array has no meanings.